Experimental/Anticipation Sample/Assets/Scripts/AnticipationSample.cs

```diff
@@ -0,0 +1,290 @@
+            GUILayout.Label("Value C (smooth, correct anticipation):");
+            updatedValue = GUILayout.HorizontalSlider(ValueC.Value, 0, 10);
+            if (updatedValue != ValueC.Value)
+            {
+                ValueC.Anticipate(updatedValue);
+                SetValueCRpc(updatedValue);
+            }
+            GUILayout.Label("Value C Current Server Value:");
+            GUILayout.HorizontalSlider(ValueC.AuthoritativeValue, 0, 10);
+            GUILayout.EndVertical();
+
+            GUILayout.BeginVertical("Box");
+            GUILayout.Label("Value D (smooth, incorrect anticipation):");
+            updatedValue = GUILayout.HorizontalSlider(ValueD.Value, 0, 10);
+            if (updatedValue != ValueD.Value)
+            {
+                ValueD.Anticipate(updatedValue);
+                SetValueDRpc(updatedValue);
+            }
+            GUILayout.Label("Value D Current Server Value:");
+            GUILayout.HorizontalSlider(ValueD.AuthoritativeValue, 0, 10);
+            GUILayout.EndVertical();
+
+            GUILayout.EndArea();
+            GUILayout.BeginArea(new Rect(610, 72, 300, 300));
+
+            GUILayout.BeginVertical("Box");
+            GUILayout.Label("Value E (Server-controlled, continuous anticipation):");
+            GUILayout.HorizontalSlider(ValueEDisplayValue % 10, 0, 10);
+            GUILayout.Label("Value E Current Server Value:");
+            GUILayout.HorizontalSlider(ValueEDisplayAuthoritative % 10, 0, 10);
+            GUILayout.EndVertical();
+
+            GUILayout.EndArea();
+            if(IsClient)
+            {
+                GUILayout.BeginArea(new Rect(0, 310, 600, 300));
+                GUILayout.Label("Anticipated Network Transform controls:");
+                GUILayout.Label("W: Move Forward | S: Move Backward | A: Turn Left | D: Turn Right");
+                GUILayout.Label("Q: Large random teleport (very different server result)");
+                GUILayout.Label("E: Small random teleport (slightly different server result)");
+                GUILayout.Label("R: Return to center (same server result)");
+                GUILayout.Label("");
+
+                GUILayout.Label($"Transform smooth duration: {Player.SmoothTime}s");
+                Player.SmoothTime = GUILayout.HorizontalSlider(Player.SmoothTime, 0, 1);
+                GUILayout.Label($"Transform smooth distance threshold: {Player.SmoothDistance}");
+                Player.SmoothDistance = GUILayout.HorizontalSlider(Player.SmoothDistance, 0, 50);
+                if (GUILayout.Button("Toggle Server Visualization (Follower)"))
+                {
+                    foreach (var childRenderer in Player.GhostTrasform.GetComponentsInChildren<MeshRenderer>())
+                    {
+                        childRenderer.enabled = !childRenderer.enabled;
+                    }
+                }
+                GUILayout.Label($"Variable smooth duration: {SmoothTime}s");
```

Comment:
Super small nitpick, but maybe we keep this slider near the variables section in the GUI?
Initially I thought this value had to do with the transform's smoothing, and not the sliders.

Reply:
Fair.

Reply:
done